lib/npm.js

```diff
@@ -295,6 +295,13 @@
           umask: umask
         }
 
+        if (config.get('proxy') && !process.env['HTTP_PROXY']){
+          process.env['HTTP_PROXY'] = config.get('proxy')
```

Comment:
Thanks for your feedback.

I would prefer only lib/utils/git.js too, but I had the impression that there are some cases where `git` is directly invoked  and it would miss the `curl` use-case.

The git.js only approach would at least make a test possible, whereas it will be almost impossible to test a general inclusion like I proposed. As there is a workaround (setting  HTTP_PROXY/HTTPS_PROXY) and thus I have no urgent need to fix it fast it might take a while.